indexer/jobs/base_job.py

```diff
@@ -207,3 +275,91 @@ def get_filter(self):
 
     def get_filter_transactions(self):
         return list(filter(self.get_filter().is_satisfied_by, self._data_buff[Transaction.type()]))
+
+
+def is_overwrite_udf(cls: Type[BaseJob]):
+    process_qualname = cls._udf.__qualname__
+    return process_qualname.startswith(cls.__name__)
+
+
+def generate_dependency_types(cls: Type[BaseJob]):
+    if not is_overwrite_udf(cls):
+        return
+
+    annotations = get_type_hints(cls._udf)
+
+    if "out" not in annotations:
```

Comment:
I think we should add a check somewhere
- output must be type of collector
- input should be valid datatypes?

Reply:
Just like u say, the input and output declarations will be checked later in the code, through the following functions "varify_input_hints" and "varify_output_hints"